lib/hcrypto/rand-w32.c

```diff
@@ -41,15 +41,16 @@
 
 #include "randi.h"
 
-volatile static HCRYPTPROV g_cryptprovider = NULL;
+#define NULL_HCRYPTPROV ((ULONG_PTR)0)
+volatile static HCRYPTPROV g_cryptprovider = NULL_HCRYPTPROV;
```

Comment:
I'm not too fussed but, I'd probably prefer not defining something that looks like it could come from the Windows headers. Just use ```(HCRYPTPROV)0``` explicitly.

Reply:
This is a global variable.   It doesn't require initialization.   It will automatically be initialized to a zero value.   Changing it to be

  volatile static HCRYPTPROV g_cryptprovider;

is sufficient.

Reply:
True. My comment applies to line 51.

Reply:
I agree.